tests/cases/compiler/numericEnumMappedType.ts

```diff
@@ -0,0 +1,38 @@
+// @strict: true
+
+// Repro from #31771
+
+enum E1 { ONE, TWO, THREE }
+declare enum E2 { ONE, TWO, THREE }
+
+type Bins1 = { [k in E1]?: string; }
+type Bins2 = { [k in E2]?: string; }
+
+const b1: Bins1 = {};
+const b2: Bins2 = {};
+
+const e1: E1 = E1.ONE;
+const e2: E2 = E2.ONE;
+
+b1[1] = "a";
+b1[e1] = "b";
+
+b2[1] = "a";
+b2[e2] = "b";
+
+// Multiple numeric enum types accrue to the same numeric index signature in a mapped type
+
+declare function val(): number;
+
+enum N1 { A = val(), B = val() }
+enum N2 { C = val(), D = val() }
+
+type T1 = { [K in N1 | N2]: K };

```

Comment:
Can you turn on `declarations: true` for this test and export this type (and Bins1 and Bins2 for good measure)? I want to see what our full structural printback for these types is.

Reply:
Sure